### PR TITLE
 Remove unused dependencies and add cargo-deny check 

### PR DESCRIPTION
**Pull request by Kijewski**

### CODE REVIEW

File: deny.toml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[licenses]
+version = 2
+allow = ["Apache-2.0", "BSD-3-Clause", "ISC", "LGPL-2.1", "MIT", "Unicode-DFS-2016"]
```
Comment (chrysn) [inline deny.toml:3]: I'd go a bit further there and limit LGPL to riot-sys.

No harm generally in GPLish licenses in the dependency tree, but I'd like to know of them. RIOT OS itself is something that is relatively straightforward to use in embedded systems with LGPL (eg. using the bindist mechanisms), but for other crates, that might be harder.

Comment (Kijewski) [reply deny.toml:3]: This is an exhaustive list of all licenses used in rust-riot-wrappers' dependency tree. `cargo deny check` marks it as a failure if at some point a new license gets added to the tree. Then you have to decide if you find this license acceptable or not.

Comment (chrysn) [reply deny.toml]: I've taken the liberty to just push what I meant: It's fine to have them here, but if we already do this check, I'd prefer to have an explicit check when a different crate than RIOT is thusly licensed.

Comment (Kijewski) [reply deny.toml]: Ahh, that's what you meant. Yes, feel free to push to my PRs. Sometimes code is easier to understand than human languages. :)

**Comment by chrysn** [comment]
This is spontaneously breaking CI when newer packages change their license.

Do you know what's best practice around that?

Is there something like "license aware cargo" (or a feature in the action) that will go for earlier versions in case the deny triggers? Or should a Cargo.lock be checked in even for a library crate just for the purpose of keeping deny happy?

**Pull request merged**